lib/client.py

```diff
@@ -54,56 +57,103 @@ def create(cls, config, username, password):
+                    headers=self.__make_headers(auth_token),
+                    method="post",
+                    data=self.__contruct_post_data(gateway_username, asset),
+                ))
+            self.__authenticator.logoff(session, self.__address_url)
+            return passwords
+
+    def get_ssh_keys(self, account, asset, gateway_username):
+        print("ssh keys")
+        with self.__requests_tls.open_session() as session:
+            auth_token = self.__authenticator.authenticate(
+                session, self.__address_url, self.__username, self.__password
+            )
+            keys = []
+            found_objects = self.__search_for_account_asset_secrets(session, auth_token, account, asset)
+            if len(found_objects) == 0:
+                logger.debug(
+                    "No keys found in vault for this account and/or asset: account=%s, asset=%s", account, asset
+                )
+                self.__authenticator.logoff(session, self.__address_url)
+                return keys
+            secret_endpoints = self.__create_secret_endpoints(self.__is_ssh_key, found_objects, account, asset)
+            for se in secret_endpoints:
+                response = session.post(
+                    urljoin(self.__address_url, se),
+                    headers=self.__make_headers(auth_token),
+                    data=json.dumps(self.__contruct_post_data(gateway_username, asset)),
+                    params=None,
+                )
+                if response.ok:
+                    keys.append(response.text.strip("\""))
+
             self.__authenticator.logoff(session, self.__address_url)
-        return passwords
+            return keys
 
-    def __get_passwords(self, session, auth_token, account, asset, gateway_username):
-        headers = {"Content-Type": "application/json", "Authorization": auth_token}
-        found_objects = self.__endpoint_extractor.extract_data_from_endpoint(
+    def __search_for_account_asset_secrets(self, session, auth_token, account, asset):
+        return self.__endpoint_extractor.extract_data_from_endpoint(
             session,
-            endpoint_url="PasswordVault/api/Accounts?search={}&sort=UserName".format(",".join([account, asset])),
+            endpoint_url=self.__search_url_template.format(",".join([account, asset])),
             data_path="value",
-            headers=headers,
+            headers=self.__make_headers(auth_token),
             method="get",
-        )
-        found_objects = list(
-            filter(lambda x: (account == x.get("userName") and asset == x.get("address")), found_objects)
-        )
-        if len(found_objects) == 0:
-            logger.debug(
-                "No objects found in vault for this account and/or asset: account={}, asset={}".format(account, asset)
+       )
+
+    @staticmethod
+    def __create_secret_endpoints(secret_type_determinant, found_objects, account, asset):
+        return list(map(
```

Comment:
might be up for debate, but I believe using list comprehension instead of list(map(filter)) pattern is much more pythonic (if the laziness of map() is not explicitly needed).
eg:
```
list(map(map_function, filter(filter_function, iterable)))
```
is more readable in this form (also much more perfomant, but that's not a great concern here):
```
[map_function(element) for element in interable if filter_function(element)]
```

Reply:
I read the filter+map combo more easily, but it's my personal preference.
I make it a comprehension :rofl: